internal/pkg/describe/lb_web_service.go

```diff
@@ -26,28 +27,30 @@ const (
 	envOutputSubdomain                 = "EnvironmentSubdomain"
 )
 
-// LBWebServiceURI represents the unique identifier to access a web service.
+// LBWebServiceURI represents the unique identifier to access a load balanced web service.
 type LBWebServiceURI struct {
-	DNSName string // The environment's subdomain if the service is served on HTTPS. Otherwise, the public load balancer's DNS.
-	Path    string // Empty if the service is served on HTTPS. Otherwise, the pattern used to match the service.
+	HTTPS    bool
+	DNSNames []string // The environment's subdomain if the service is served on HTTPS. Otherwise, the public load balancer's DNS.
+	Path     string   // Empty if the service is served on HTTPS. Otherwise, the pattern used to match the service.
 }
 
-func (uri *LBWebServiceURI) String() string {
-	switch uri.Path {
-	// When the service is using host based routing, the service
-	// is included in the DNS name (svc.myenv.myproj.dns.com)
-	case "":
-		return fmt.Sprintf("https://%s", uri.DNSName)
-	// When the service is using the root path, there is no "path"
-	// (for example http://lb.us-west-2.amazon.com/)
-	case "/":
-		return fmt.Sprintf("http://%s", uri.DNSName)
-	// Otherwise, if there is a path for the service, link to the
-	// LoadBalancer DNS name and the path
-	// (for example http://lb.us-west-2.amazon.com/svc)
-	default:
-		return fmt.Sprintf("http://%s/%s", uri.DNSName, uri.Path)
+func (u *LBWebServiceURI) String() string {
+	var uris []string
+	for _, dnsName := range u.DNSNames {
+		uri := ""
+		switch strconv.FormatBool(u.HTTPS) + "|" + strconv.FormatBool(u.Path == "/") {
+		case "true|true":
+			uri = fmt.Sprintf("https://%s", dnsName)
+		case "false|true":
+			uri = fmt.Sprintf("http://%s", dnsName)
+		case "true|false":
+			uri = fmt.Sprintf("https://%s/%s", dnsName, u.Path)
+		case "false|false":
+			uri = fmt.Sprintf("http://%s/%s", dnsName, u.Path)
+		}
+		uris = append(uris, uri)
 	}
+	return strings.Join(uris, " or ")
```

Comment:
can we use: https://github.com/dustin/go-humanize#word-series ?

Reply:
hmm why do we need this?

Reply:
oooh i see nvm